MatrixSDK/Data/MXRoom.m

```diff
@@ -1427,14 +1427,34 @@ - (MXHTTPOperation*)sendAudioFile:(NSURL*)fileLocalURL
 
 - (MXHTTPOperation*)sendVoiceMessage:(NSURL*)fileLocalURL
                             mimeType:(NSString*)mimeType
+                            duration:(NSTimeInterval)duration
+                             samples:(NSArray<NSNumber *> *)samples
                            localEcho:(MXEvent**)localEcho
                              success:(void (^)(NSString *eventId))success
                              failure:(void (^)(NSError *error))failure
                   keepActualFilename:(BOOL)keepActualName
 {
+    NSMutableDictionary *extensibleAudioContent = @{kMXMessageContentKeyExtensibleAudioDuration : @(duration)}.mutableCopy;
+ 
+    static NSUInteger scaledWaveformSampleCeiling = 1024;
+    
+    NSMutableArray *scaledSamples = [NSMutableArray array];
+    for (NSNumber *sample in samples) {
```

Comment:
Is this just for a small number of averaged samples that have already been prepared between 0 and 1 for rendering a basic waveform?